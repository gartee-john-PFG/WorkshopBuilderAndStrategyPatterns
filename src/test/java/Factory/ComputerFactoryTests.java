package Factory;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

public class ComputerFactoryTests {
    @Test
    public void testPCCreation() {
        Computer computer = ComputerFactory.getComputer("PC", "32 GB", "1 TB SSD", "Core i9");

        Approvals.verify(computer.toString());
    }

    @Test
    public void testServerCreation(){
        Computer computer = ComputerFactory.getComputer("Server", "64 GB", "4 x 4 TB SSD", "Xeon i9");

        Approvals.verify(computer.toString());

    }
}
