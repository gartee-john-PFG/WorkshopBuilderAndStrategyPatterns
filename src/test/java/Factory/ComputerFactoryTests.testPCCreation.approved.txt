Type: PC, RAM = 32 GB, HDD = 1 TB SSD, CPU = Core i9